# EDA Smartphones

### Librerías

In [1]:
import pandas as pd
from utils import funciones as f

# pd.set_option('display.max_colwidth', None)

### Data

In [2]:
df = pd.read_csv('.\data\smartphonesmessy.csv')

In [3]:
df.shape

(1020, 11)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1020 entries, 0 to 1019
Data columns (total 11 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   model      1020 non-null   object 
 1   price      1020 non-null   object 
 2   rating     879 non-null    float64
 3   sim        1020 non-null   object 
 4   processor  1020 non-null   object 
 5   ram        1020 non-null   object 
 6   battery    1020 non-null   object 
 7   display    1020 non-null   object 
 8   camera     1019 non-null   object 
 9   card       1013 non-null   object 
 10  os         1003 non-null   object 
dtypes: float64(1), object(10)
memory usage: 87.8+ KB


In [5]:
df.describe(include = 'all').round()

model    price  rating                             sim  \
count            1020     1020   879.0                            1020   
unique           1020      412     NaN                              28   
top     OnePlus 11 5G  ₹14,999     NaN  Dual Sim, 3G, 4G, VoLTE, Wi-Fi   
freq                1       21     NaN                             324   
mean              NaN      NaN    78.0                             NaN   
std               NaN      NaN     7.0                             NaN   
min               NaN      NaN    60.0                             NaN   
25%               NaN      NaN    74.0                             NaN   
50%               NaN      NaN    80.0                             NaN   
75%               NaN      NaN    84.0                             NaN   
max               NaN      NaN    89.0                             NaN   

                                             processor  \
count                                             1020   
unique                                             298   
top     Dimensity 700 5G, Octa Core, 2.2 GHz Processor   
freq                                                29   
mean                                               NaN   
std                                                NaN   
min                                                NaN   
25%                                                NaN   
50%                                                NaN   
75%                                                NaN   
max                                                NaN   

                             ram                                  battery  \
count                       1020                                     1020   
unique                        58                                      256   
top     8 GB RAM, 128 GB inbuilt  5000 mAh Battery with 33W Fast Charging   
freq                         267                                      103   
mean                         NaN                                      NaN   
std                          NaN                                      NaN   
min                          NaN                                      NaN   
25%                          NaN                                      NaN   
50%                          NaN                                      NaN   
75%                          NaN                                      NaN   
max                          NaN                                      NaN   

                                                  display  \
count                                                1020   
unique                                                369   
top     6.67 inches, 1080 x 2400 px, 120 Hz Display wi...   
freq                                                   54   
mean                                                  NaN   
std                                                   NaN   
min                                                   NaN   
25%                                                   NaN   
50%                                                   NaN   
75%                                                   NaN   
max                                                   NaN   

                                                   camera  \
count                                                1019   
unique                                                285   
top     50 MP + 8 MP + 2 MP Triple Rear & 16 MP Front ...   
freq                                                   40   
mean                                                  NaN   
std                                                   NaN   
min                                                   NaN   
25%                                                   NaN   
50%                                                   NaN   
75%                                                   NaN   
max                                                   NaN   

                                    car

In [6]:
f.cardinalidad(df)

variable tipo_dato  cantidad_de_nulos  \
0       model    object                  0   
1       price    object                  0   
2      rating   float64                141   
3         sim    object                  0   
4   processor    object                  0   
5         ram    object                  0   
6     battery    object                  0   
7     display    object                  0   
8      camera    object                  1   
9        card    object                  7   
10         os    object                 17   

                                       valores_unicos  cardinalidad  \
0   [OnePlus 11 5G, OnePlus Nord CE 2 Lite 5G, Sam...          1020   
1   [₹54,999, ₹19,989, ₹16,499, ₹14,999, ₹24,999, ...           412   
2   [89.0, 81.0, 75.0, 82.0, 80.0, 86.0, 85.0, 84....            30   
3   [Dual Sim, 3G, 4G, 5G, VoLTE, Wi-Fi, NFC, Dual...            28   
4   [Snapdragon 8 Gen2, Octa Core, 3.2 GHz Process...           298   
5   [12 GB RAM, 256 GB inbuilt, 6 GB RAM, 128 GB i...            58   
6   [5000 mAh Battery with 100W Fast Charging, 500...           256   
7   [6.7 inches, 1440 x 3216 px, 120 Hz Display wi...           369   
8   [50 MP + 48 MP + 32 MP Triple Rear & 16 MP Fro...           285   
9   [Memory Card Not Supported, Memory Card (Hybri...            63   
10  [Android v13, Android v12, iOS v16, No FM Radi...            48   

    porcentaje_cardinalidad tipo_variable  
0                100.000000       nominal  
1                 40.392157      continua  
2                  2.941176      discreta  
3                  2.745098       nominal  
4                 29.215686       nominal  
5                  5.686275       ordinal  
6                 25.098039       ordinal  
7                 36.176471       nominal  
8                 27.941176       ordinal  
9                  6.176471       ordinal  
10                 4.705882       nominal

##### Al analizar los valores dentro de ciertas columnas podemos concluir:
- La columna 'ram' tiene datos que no corresponden a por ejemplo: 'Dual Core, 500\u2009MHz Processor'
- La columna 'processor' con tiene datos que no se corresponden los demás por ejemplo: '4\u2009MB RAM, 4\u2009MB inbuilt', '1450\u2009mAh Battery'
- La columna 'ram' tiene datos que no se corresponden con lo demás, por ejemplo: '1.5 inches, 120\u2009x\u2009120\u2009px Display'
- La columna 'battery' tiene datos que no se corresponden con los demás, por ejemplo: '48\u2009MB RAM, 128\u2009MB inbuilt'
- La columna 'display' tiene datos que no se corresponden con los demás, por ejemplo: '0.3\u2009MP Rear Camera'
- La columna 'camera' tiene datos que no se corresponden con los demás, por ejemplo: '1.77 inches, 240\u2009x\u2009320\u2009px Display'
- La columna 'card' tiene datos que no se corresponden con los demás, por ejemplo: '50\u2009MP + 48\u2009MP + 32\u2009MP Triple Rear & 32\u2009MP + 32\u2009MP Dual Front Camera'
- La columna 'os' tiene datos que no se corresponden con los demás, por ejemplo: 'Memory Card Supported, upto 32\u2009GB'
- La columna 'price' esta en rupias